### PR TITLE
OptionsPanel tests, fix clip button enablement render in PDF mode, cleanup

### LINKED ISSUE #183: clipButtonEnabled() returns wrong values for PDF mode
Encountered this when I was writing UTs:

1) All Pages selected, something invalid was typed into selection text box
Expected: true, but was false
2) localFilesAllowed is a deceiving name, should be renamed to isLocalFileAndNotAllowed or similar

### PR DESCRIPTION
**Pull request by mttwc**
Added a bunch of UTs for the rendering of an enabled/disabled clip button in optionsPanel for each of our modes. While I was testing this, I noticed that we had incorrect logic for this very thing, and it's now corrected. Also cleaned up some code.

Fixes #183 and #179

### CODE REVIEW

File: src/scripts/clipperUI/clipperStateUtilities.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+				if (!clipperState.pdfPreviewInfo.isLocalFileAndNotAllowed) {
+					return false;
+				} else if (clipperState.pdfResult.status !== Status.Succeeded) {
+					return false;
+				} else if (clipperState.pdfPreviewInfo.allPages) {
+					return true;
+				}
+				// We know at this point that the status is succeeded and the user has a specific page range, so if
+				// it is unparseable or empty, we disallow the clip
+				let pages = StringUtils.parsePageRange(clipperState.pdfPreviewInfo.selectedPageRange);
```
Comment (leakymemory) [inline src/scripts/clipperUI/clipperStateUtilities.ts]: What about the extra parameter which passes the total number of pages?  Without it we won't be able to validate the upper page limit, no?

**Review by leakymemory — APPROVED** [review]
Just the one question, but this looks good!

File: src/scripts/clipperUI/clipperStateUtilities.ts (hunk 1 above)
Comment (mttwc) [reply src/scripts/clipperUI/clipperStateUtilities.ts]: Oh yeah, that's true. I'll add a change and add a UT for this case.

**Comment by mttwc** [comment]
Resolved Chad's comment by adding back the upper bounds check and added a matching UT.

**Pull request merged**